cfme/tests/infrastructure/test_tenant_quota.py

```diff
@@ -238,3 +272,49 @@ def test_setting_child_quota_more_than_parent(tenants_setup, parent_quota, child
     view.flash.assert_message('Error when saving tenant quota: Validation failed: {} allocated '
                               'quota is over allocated, parent tenant does not have enough quota'.
                               format(flash_text))
+
+
+@pytest.mark.parametrize(
+    ['parent_quota', 'child_quota', 'flash_text'],
+    [
+        [('cpu', '5'), ('cpu', '3'), 'Cpu'],
+        [('memory', '5'), ('memory', '3'), 'Mem'],
+        [('storage', '5'), ('storage', '3'), 'Storage'],
+        [('vm', '5'), ('vm', '3'), 'Vms'],
+        [('template', '5'), ('template', '3'), 'Templates']
+    ],
+    ids=['cpu', 'memory', 'storage', 'vm', 'template']
+)
+def test_vm_migration_after_assigning_tenant_quota(appliance, new_vm, tenant_quota, provider):
+    """prerequisite: Provider should be added
+
+    steps:
+
+    1. Create VM
+    2. Assign tenant quota
+    3. Migrate VM
+    4. Check whether migration is successfully done
+
+    :param appliance:
+    :param new_vm:
+    :param provider:
+    :param tenants_setup:
+    :param parent_quota:
+    :param child_quota:
+    :param flash_text:
+    """
+    view = navigate_to(new_vm, 'Details')
```

Comment:
Required: move this to the fixture:
def check_hosts(new_vm, provider):
if  len(provider.hosts) != 1:
view = navigate_to(new_vm, 'Details')
vm_host = view.entities.summary('Relationships').get_text_of('Host')
hosts = [vds.name for vds in provider.hosts if vds.name not in vm_host]
return hosts[0]
else:
pytest.skip("There is only one host in the provider")